### PR TITLE
Add storages operation for managedEnvironment

### PR DESCRIPTION
**Pull request by Juliehzl**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by howang-ms — APPROVED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ManagedEnvironmentsStorages.json [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/managedEnvironments/{envName}/storages": {
+      "get": {
+        "tags": [
+          "ManagedEnvironmentsStorages"
+        ],
+        "summary": "Get all storages for a managedEnvironment.",
+        "description": "GGet all storages for a managedEnvironment.",
```
Comment (ruslany) [inline specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ManagedEnvironmentsStorages.json:24]: >GGet [](http://example.com/codeflow?start=24&length=4)

Typo

File: specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ManagedEnvironmentsStorages.json [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "List environments storages by subscription": {
+            "$ref": "./examples/ManagedEnvironmentsStorages_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/managedEnvironments/{envName}/storages/{name}": {
```
Comment (raosuhas) [inline specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ManagedEnvironmentsStorages.json:65]: >storages [](http://example.com/codeflow?start=126&length=8)

consider updating the name. "storages" is not really a word. Something like "stores" or "managedstores" (These are just suggestions please work with your team to settle on a proper name)

File: specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/examples/ManagedEnvironmentsStorages_CreateOrUpdate.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          "shareName": "share1",
+          "accessMode": "ReadOnly"
+        }
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/8efdecc5-919e-44eb-b179-915dca89ebf9/resourceGroups/examplerg/providers/Microsoft.App/managedEnvironments/managedEnv/storages/jlaw-demo1",
```
Comment (raosuhas) [inline specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/examples/ManagedEnvironmentsStorages_CreateOrUpdate.json:22]: Hey, can you please just explain the scenario a bit here ?

The customer creates the storages passing in the storageaccount key and then you return this resource with the file details. 

- How is this information used by your service ?
- What happens if the storage account key rotates ?

Have you considered using managed identity and accessing the storage using that route ?

Comment (howang-ms) [reply specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/examples/ManagedEnvironmentsStorages_CreateOrUpdate.json:22]: The storage resource will be reference by the containerapps resource by name. And the service will mount the storage to the container when it is referenced. It is 1:N mapping, so a single file share can be mount to multiple containerapps.
When the account key is rotates, the customer need to update the storage with the latest account information.

For long term, we will support both managed identity and access key based access. But for GA, we will only support access key.

File: specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ManagedEnvironmentsStorages.json (hunk 2 above)
Comment (howang-ms) [reply specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ManagedEnvironmentsStorages.json:65]: https://en.wiktionary.org/wiki/storages#Englis
https://www.wordhippo.com/what-is/the-plural-of/storage.html

I think storage have a "plural" as "storages". What do you think?

File: specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ManagedEnvironmentsStorages.json (hunk 1 above)
Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ManagedEnvironmentsStorages.json]: fixed

File: specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ManagedEnvironmentsStorages.json [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+            "in": "path",
+            "description": "Name of the Environment.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
```
Comment (pilor) [inline specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/ManagedEnvironmentsStorages.json:44]: >responses [](http://example.com/codeflow?start=9&length=9)

needs to be marked x-ms-pageable  [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi009-always-add-x-ms-pageable-to-list-calls](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi009-always-add-x-ms-pageable-to-list-calls)

**Comment by pilor** [comment]
signing off with the 1 remaining comment

**Review by leniatgh — APPROVED** [review]

**Pull request merged**